### DISCUSSION TITLE
[concept] Full featured dedicated forum

### DISCUSSION THREAD

**Discussion by fggxf** [element-hq/element-meta#481] [Ideas]
**Idea:**
Having a built in full featured dedicated forum with attachment support would be great, it would be better structured, conducive to more long term serious discussions and would prevent your post from getting railroaded by a steam of sh*tposting in the servers IM stream.

The forum view would be toggled to either replace or splitview with the IM stream

**Integration with the IM stream:**
 Topic creation notification in the servers instant message stream when a user creates a new topic, like a normal server message.
 Users could have the ability to click the notification to open the split view and display the topic.

really dirty concept/mock-up image:
![dirty mock2](https://user-images.githubusercontent.com/42724880/55641820-b32e5080-5783-11e9-83a2-028c121151f3.png)

I feel like this is a major feature missing from modern chat applications and makes them feel  chaotic when multiple people are trying to have separate discussions.

**Comment by turt2live** [comment]
Threading ( https://github.com/vector-im/riot-web/issues/2349 ) might also solve this depending on how it is designed. It could result in letting the noise fall out of sight and the important stuff being easy to find.

**Comment by fggxf** [comment]
I think this would be the better approach overall because all the servers would each have a dedicated forum... with categories, sub-forums and threads. We could also just have both, threading for quick replies/threads that don't warrant a forum post and the forum for full-blown long-term discussions spanning hundreds/thousands of pages over many years. The forum post themselves could even be threaded/nested.
This way the servers could become destinations and feel even more like a community. This also has the possibility of rendering traditional forums obsolete because it would be more secure, private and practically push button simple.

**Comment by erlend-sh** [comment]
This is quite doable now that Matrix supports threads through label filters: https://github.com/matrix-org/synapse/pull/6301

Being able to bridge to an established forum API like Discourse or GitHub Discussions would be a major first step.

**Comment by schrmh** [comment]
I'd also like to see forums.
Especially since one competitor started testing a feature like that:
![image](https://user-images.githubusercontent.com/32062480/174525822-2dfdce7d-dedc-4494-b2f6-e9921f1b5217.png)
![image](https://user-images.githubusercontent.com/32062480/174525845-b32ed645-7fcf-4372-b77a-a18b5a8d0a3e.png)
![image](https://user-images.githubusercontent.com/32062480/174525893-a934861a-8563-4569-a9a3-b5606135c617.png)
![image](https://user-images.githubusercontent.com/32062480/174525907-9af5f77b-398a-440c-a914-223e474ce673.png)
